presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1100,6 +1102,16 @@ public SystemSessionProperties(
                         AGGREGATION_IF_TO_FILTER_REWRITE_ENABLED,
                         "Enable rewriting the IF expression inside an aggregation function to a filter clause outside the aggregation",
                         featuresConfig.isAggregationIfToFilterRewriteEnabled(),
+                        false),
+                booleanProperty(
+                        HEAP_DUMP_ON_EXCEEDED_MEMORY_LIMIT_ENABLED,
+                        "Trigger heap dump to `EXCEEDED_MEMORY_LIMIT_HEAP_DUMP_FILE_PATH` on exceeded memory limit exceptions",
+                        false,
```

Comment:
for the default value, we usually specify it in a config; for this case, `featuresConfig` would be a good one. Check the other examples in this class.

Reply:
When I considered it I decided that the way it is structured it doesn't make much sense as a config option. It's a pretty niche usage and it will only dump the heap once per process (there is a guard in heap dumper).

I'm not sure how applicable this is for production clusters given that restriction. It seems like a good restriction just because it limits the blast radius of accidentally enabling the option and blasting it at a production cluster (yikes!).

Just thinking about it makes me think we actually want to prohibit this by default unless the cluster is started with an option permitting it that can't be set via session property (so features config).

Reply:
Also, add a comment to these two places to explain the rationale why we don't want a config